src/linker/Linker/MessageCategory.cs

```diff
@@ -6,11 +6,11 @@ namespace Mono.Linker
 {
 	public enum MessageCategory
 	{
-		Error = 0,
-		Warning,
-		Info,
-		Diagnostic,
-
-		WarningAsError = 0xFF
+		Error = 1,
```

Comment:
The `All` value doesn't make sense anymore. I would just revert this entire file.